src/main/scala/com/microsoft/hyperspace/index/sources/interfaces.scala

```diff
@@ -222,13 +201,54 @@ trait FileBasedSourceProvider extends SourceProvider {
   def getRelation(plan: LogicalPlan): Option[FileBasedRelation]
 
   /**
-   * Returns enriched index properties.
+   * Returns true if the given relation metadata is supported relation metadata.
    *
-   * @param relation Logical relation to retrieve necessary information.
-   * @param properties Index properties to enrich.
-   * @return Updated index properties for index creation or refresh.
+   * @param metadata Relation metadata to check if it's supported.
+   * @return Some(true) if the given relation metadata is supported relation metadata, otherwise
+   *         None.
+   */
+  def isSupportedRelationMetadata(metadata: Relation): Option[Boolean]
+
+  /**
+   * Returns the [[FileBasedRelationMetadata]] that wraps the given relation metadata
+   * if the given relation metadata is supported relation metadata
+   *
+   * @param metadata Relation metadata to wrap to [[FileBasedRelationMetadata]]
+   * @return [[FileBasedRelationMetadata]] that wraps the given relation metadata
+   */
+  def getRelationMetadata(metadata: Relation): Option[FileBasedRelationMetadata]
+}
+
+/**
+ * ::Experimental::
+ * A trait that represents relation metadata for a source provider.
+ *
+ * @since 0.5.0
+ */
+trait SourceRelationMetadata
+
+/**
+ * ::Experimental::
+ * A trait that a source provider should implement to represent the source relation metadata
+ * that is based on files.
+ *
+ * @since 0.5.0
+ */
+trait FileBasedRelationMetadata extends SourceRelationMetadata {
+
+  /**
+   * @return new [[Relation]] metadata that will have the latest source
+   */
+  def refresh(): Relation
+
+  /**
+   * @return file format name to read internal data
+   */
+  def internalFileFormatName(): String
+
+  /**
+   * @param properties Index properties to enrich
```

Comment:
Could you add some function comments like getRelationMetadata? as this is a kind of public interface.